### PR TITLE
making new env multi

### PR DESCRIPTION
**Pull request by memsharded**
Changelog: Feature: Make new environment generators multi-config (Release/Debug and arch).
Changelog: Fix: Make automatic call of ``VirtualBuildEnv`` and ``VirtualRunEnv`` independent.
Docs: https://github.com/conan-io/docs/pull/2247

Close https://github.com/conan-io/conan/pull/9443/

#tags: slow

### CODE REVIEW

File: conans/test/functional/toolchains/env/test_complete.py [hunk 1]
```diff
@@ -122,18 +122,22 @@ class Pkg(ConanFile):
...
             generators = "CMakeDeps", "CMakeToolchain", "VirtualBuildEnv", "VirtualRunEnv"
+            apply_env = False
 
             def build(self):
                 cmake = CMake(self)
                 cmake.configure()
                 cmake.build()
                 self.run("mycmake")
                 self.output.info("RUNNING MYAPP")
+                filename = "conanrunenv-{}-{}".format(self.settings.build_type,
```
Comment (lasote) [inline conans/test/functional/toolchains/env/test_complete.py]: Not a good UI,  Should we enable all the `env` collaborators a common interface to get the name? something like `VirtualRunEnv(self).filename`,  `VCVars(self).filename`, ... 
The bad part is that you need to import the class.

**Review by lasote — APPROVED** [review]
I never liked much to have to recall the filename to activate the environment, but now compose it with the settings I think is too much. But overall is good.

File: conans/test/functional/toolchains/env/test_complete.py (hunk 1 above)
Comment (memsharded) [reply conans/test/functional/toolchains/env/test_complete.py]: But I have made it configurable, in the ``generate()`` method you can change that (same pattern as CMakeDeps)
So you cannot create a new instance, we would need to serialize somewhere the name of the file, and that would be ugly too...

Comment (memsharded) [reply conans/test/functional/toolchains/env/test_complete.py]: I have rewritten the whole ``env_scripts`` approach, now can be used for build, run (and potentially other environments at the user control), so this solves this UX issue. It was necessary some breaking changes, please review.

File: conan/tools/env/environment.py [hunk 2]
```diff
@@ -427,6 +427,7 @@ def loads(text):
         return result
 
 
-def register_environment_script(conanfile, path):
-    if path not in conanfile.environment_scripts:
-        conanfile.environment_scripts.append(path)
+def register_env_script(conanfile, path, group):
```
Comment (franramirez688) [inline conan/tools/env/environment.py]: Please, add a docstring explaining a little bit the function and its parameters.

**Review by lasote — APPROVED** [review]
Good idea! now is much more comfortable.

**Review by franramirez688 — APPROVED** [review]

File: conans/test/functional/toolchains/gnu/autotools/test_win_bash.py [hunk 3]
```diff
@@ -67,11 +67,13 @@ def build(self):
                  "main.cpp": main})
     client.run("install .")
     client.run("build .")
+    print(client.out)
+    print(client.current_folder)
```
Comment (czoido) [inline conans/test/functional/toolchains/gnu/autotools/test_win_bash.py]:
```suggestion
```

**Review by czoido — APPROVED** [review]

**Pull request merged**